### PR TITLE
suggested io changes

### PR DESCRIPTION
**Pull request by jannes-m**
just started with the review, so there are more commits to come but this way you already can keep track of what I am doing

### CODE REVIEW

**Comment by Robinlovelace** [comment]
OK great stuff, many thanks for this. Will 'hold fire' and allow @Nowosad to look over these before merging. This is definitely good stuff!

File: 05-read-write-plot.Rmd [hunk 1]
```diff
@@ -2,61 +2,59 @@
 
 The previous chapters provided an overview of spatial data classes in R, with a focus on simple features and rasters.
 This chapter is about getting spatial data onto your computer and then, perhaps after processing it with techniques described in this book, back out to the world.
-Geocomputation usually involves reading existing data, and saving it after processing sections.
-Therefore, it is important be aware of the most often used spatial file format and their strong and weak sides (see section \@ref(file-formats)).
+Geocomputation usually involves reading existing data, and saving it after some processing.
```
Comment (Robinlovelace) [inline 05-read-write-plot.Rmd]: Yes this is a definitely clearer. Thanks for that.

File: 05-read-write-plot.Rmd [hunk 2]
```diff
@@ -421,22 +419,22 @@ saveWidget(mapview_obj, file="my_interactive_map.html")
 
 1. List and describe three types of vector, raster, and geodatabase formats.
 
-1. Name at least differences between `read_sf()` and the more well-known function `st_read()`.
+2. Name at least two differences between `read_sf()` and the more well-known function `st_read()`.
```
Comment (Robinlovelace) [inline 05-read-write-plot.Rmd]: It does not matter that they are always 1s - Markdown knows what to do: https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#lists

File: 05-read-write-plot.Rmd [hunk 3]
```diff
@@ -421,22 +419,22 @@ saveWidget(mapview_obj, file="my_interactive_map.html")
...
-1. Create a static map of Germany borders and save it to a PNG file.
+6. Create a static map of Germany's borders, and save it to a PNG file.
 
-1. Create an interactive map using data from the `cycle_hire_xy.csv` file. 
-Export this map to the HTML file called `cycle_hire.html`.
+7. Create an interactive map using data from the `cycle_hire_xy.csv` file. 
+Export this map to a file called `cycle_hire.html`.
 
-[^1]: A list of supported vector formats and theirs options could be found at http://www.gdal.org/ogr_formats.html.
+[^1]: A list of supported vector formats and theirs options can be found at http://www.gdal.org/ogr_formats.html.
```
Comment (Robinlovelace) [inline 05-read-write-plot.Rmd]: Yes this is definitely right - alongside your other changes these are great improvements.

File: 05-read-write-plot.Rmd [hunk 4]
```diff
@@ -1,62 +1,167 @@
...
 
 I/O is short for "input/output" which means, in plain English, "reading and writing data".
-We use the acronym instead of plain English not to confuse you or to make chapter names short, but because that's the term used in computer science and it is useful to think of data import and export from a computing perspective.^[
+We use the acronym instead of plain English not to confuse you or to make chapter names short, but because that is the term used in computer science, and it is useful to think of data import and export from a computing perspective.^[
 Concepts such as computational efficiency, hard disk space and 'idempotence' are useful when thinking about reading and writing geographic datasets, which can become large and difficult to handle.
 Loading/saving data is yet another way of saying the same thing.
 ]
 
-## File formats
+## Retrieving open data {#retrieving-data}
```
Comment (jannes-m) [inline 05-read-write-plot.Rmd]: I would start with retrieving data since this is the natural first step before doing anything else such as importing it.

File: 05-read-write-plot.Rmd [hunk 5]
```diff
@@ -1,62 +1,167 @@
...
+Nowadays, a vast amount of spatial data is available on the internet.
```
Comment (jannes-m) [inline 05-read-write-plot.Rmd]: Here, I have just provided a little bit of context (geolibraries, geoportals and corresponding examples).

File: 05-read-write-plot.Rmd [hunk 6]
```diff
@@ -150,106 +253,22 @@ raster_filepath = system.file("raster/srtm.tif", package = "spDataLarge")
...
-single_layer_band3 = raster(raster_multilayer_filepath, band = 3)
-```
-
-The whole multilayer raster file is read by `brick()`.
-
-```{r}
-multilayer_layer_brick = brick(raster_multilayer_filepath)
-```
-
-Finally, many single and multilayer rasters could be read together using the `stack()` function:
```
Comment (jannes-m) [inline 05-read-write-plot.Rmd]: I deleted this part because we showed similar code in c2. But still there is a problem, in c2 we are already presenting how to read in raster data using raster, stack and brick. And then explain what the differences are. Hence, I would suggest to just show in c2 how to manually create a raster and how to stack two rasters while referring to c5 for a more detailed explanation regarding raster stacks and bricks (i.e. move great parts of the raster classes section to c5). What do you think?

File: 05-read-write-plot.Rmd [hunk 7]
```diff
@@ -407,36 +428,36 @@ save_tmap(tm  = tmap_obj, filename = "lifeExp_tmap.png")
 <!-- a = plot(world["lifeExp"]) -->
 <!-- ``` -->
 
-On the other hand, an interactive map could be usually saved as an HTML file. 
-This can be done with the `saveWidget()` function from the `htmlwidget` package:
+On the other hand, on can save interactive maps as an HTML file using the `saveWidget()` function from the `htmlwidgets` package:
 
+<!-- example doesn't work, problem with colors I guess -->
```
Comment (jannes-m) [inline 05-read-write-plot.Rmd]: the mapview example doen't work, pls fix

File: 05-read-write-plot.Rmd [hunk 8]
```diff
@@ -150,109 +260,26 @@ raster_filepath = system.file("raster/srtm.tif", package = "spDataLarge")
...
-world_raw_filepath = system.file("shapes/world.gpkg", package = "spData")
-world_raw = st_read(world_raw_filepath)
-```
-
-More information on getting data using R packages could be found in [section 5.5](https://csgillespie.github.io/efficientR/input-output.html#download) and [section 5.6](https://csgillespie.github.io/efficientR/input-output.html#accessing-data-stored-in-packages) of @gillespie_efficient_2016.
+Please refer to section \@ref(raster-classes) for information on the difference between raster stacks and bricks.
 
 ## Data output (O) {#data-output}
 
+<!--maybe we can come up withan intro which is a bit more compelling-->
```
Comment (jannes-m) [inline 05-read-write-plot.Rmd]: I think we can write an intro which is a bit more compelling. I have to run now, so maybe you can come up with sth. or else I will think about it later on.

**Comment by jannes-m** [comment]
For the moment, I am done with reviewing c5, so feel free to merge.

File: 05-read-write-plot.Rmd (hunk 7 above)
Comment (Nowosad) [reply 05-read-write-plot.Rmd]: That's a bug in mapview. It is already fixed in the dev version on gh, so it should be on CRAN soon.
